src/data/csvloader.cpp

```diff
@@ -0,0 +1,196 @@
+        ));
+        return Dataset();
+    }
+
+    Dataset res;
+    res.entries.reserve(rows.size() - 1);
+
+    // First row is the column names
+    const std::vector<std::string>& columns = rows.front();
+
+    int xColumn = -1;
+    int yColumn = -1;
+    int zColumn = -1;
+
+    int nDataColumns = 0;
+    bool hasExcludeColumns = specs.has_value() && (*specs).hasExcludeColumns();
+    std::vector<size_t> skipColumns;
+    if (hasExcludeColumns) {
+        skipColumns.reserve((*specs).excludeColumns.size());
+    }
+
+    for (size_t i = 0; i < columns.size(); ++i) {
+        const std::string& col = columns[i];
+
+        if (isPositionColumn(col, specs)) {
+            if (isColumnX(col, specs)) {
+                xColumn = static_cast<int>(i);
+            }
+            if (isColumnY(col, specs)) {
+                yColumn = static_cast<int>(i);
+            }
+            if (isColumnZ(col, specs)) {
+                zColumn = static_cast<int>(i);
+            }
+        }
+        else if (hasExcludeColumns && (*specs).isExcludeColumn(col)) {
+            skipColumns.push_back(i);
+            continue;
+        }
+        else {
+            res.variables.push_back({
+                .index = nDataColumns,
+                .name = col
+            });
+            nDataColumns++;
+        }
+    }
+
+    if (xColumn < 0 || yColumn < 0 || zColumn < 0) {
+        // One or more position columns weren't read
+        LERROR(fmt::format(
+            "Error loading data file {}. Missing X, Y or Z position column", filePath
+        ));
+    }
+
+    LINFO(fmt::format(
+        "Loading {} rows with {} columns", rows.size(), columns.size()
+    ));
+    ProgressBar progress(rows.size());
+
+    // Skip first row (column names)
+    for (size_t rowIdx = 1; rowIdx < rows.size(); ++rowIdx) {
+        const std::vector<std::string>& row = rows[rowIdx];
+
+        Dataset::Entry entry;
+        entry.data.reserve(nDataColumns);
+
+        for (size_t i = 0; i < row.size(); ++i) {
+            // Check if column should be exluded. Note that list of indices is sorted
+            // so we can do a binary search
+            if (hasExcludeColumns &&
+                std::binary_search(skipColumns.begin(), skipColumns.end(), i))
+            {
+                continue;
+            }
+
+            const std::string& strValue = row[i];
+
+            // For now, all values are converted to float
+            float value = readFloatData(strValue);
+
+            if (i == xColumn || i == yColumn || i == zColumn) {
+                if (i == xColumn) {
+                    entry.position.x = value;
+                }
+                if (i == yColumn) {
+                    entry.position.y = value;
+                }
+                if (i == zColumn) {
+                    entry.position.z = value;
+                }
+            }
+            else {
+                entry.data.push_back(value);
+            }
```

Comment:
```suggestion
            if (i == xColumn) {
                entry.position.x = value;
            }
            else if (i == yColumn) {
                entry.position.y = value;
            }
            else if (i == zColumn) {
                entry.position.z = value;
            }
            else {
                entry.data.push_back(value);
            }
```

?

Reply:
I had it like this and then rewrote it for some reason.. I wonder why 😅 (December brain, maybe...)

But yes, it should be the same